test/inductor/test_torchinductor.py

```diff
@@ -7637,6 +7637,26 @@ def fn(arg3_1, arg3_2, relu, permute_1):
             ref = torch.compile(fn, fullgraph=True)(*args)
             assert same(ref, correct)
 
+        @requires_cuda()
+        def test_deterministic_algorithms(self):
+            N = 10000
+
+            @torch.compile
+            def fn(idx, values):
+                x = torch.zeros(1, device="cuda")
+                x[idx] += values
+                return x
+
+            idx = torch.zeros(N, dtype=torch.int64, device="cuda")
+            values = torch.randn(N, device="cuda")
+
+            r0 = fn(idx, values)
+            with DeterministicGuard(True):
+                r1 = fn(idx, values)
+                for _ in range(10):
+                    rn = fn(idx, values)
+                    assert (r1 == rn).all()
```

Comment:
should it be `self.assertEqual` with 0 tolerance?

Reply:
I'll update it.

Is there a functional difference? (I originally used `assert` because that's what was used by the nearby test cases.)

Reply:
No functional difference, TestCase.assertEqual provides more detailed error message when it doesn't pass (maximum difference, index of the maximum difference etc).